src/EFCore.MySql/Infrastructure/MySqlDbContextOptionsBuilder.cs

```diff
@@ -31,6 +31,12 @@ public virtual MySqlDbContextOptionsBuilder ServerVersion(Version version, Serve
         public virtual MySqlDbContextOptionsBuilder ServerVersion(string serverVersion)
             => WithOption(e => e.WithServerVersion(new ServerVersion(serverVersion)));
 
+        /// <summary>
+        ///     Configures the Default CharSet
+        /// </summary>
+        public virtual MySqlDbContextOptionsBuilder CharSetConfig(CharSetConfig charSetConfig)
```

Comment:
Instead of taking a mutable options object we prefer to present a Fluent API builder, see [WarningsConfigurationBuilder.cs](https://github.com/aspnet/EntityFrameworkCore/blob/5b894d9edf8aecd240629a4ee6149470353a65f1/src/EFCore/Diagnostics/WarningsConfigurationBuilder.cs) and [WarningsConfiguration.cs](https://github.com/aspnet/EntityFrameworkCore/blob/280f2ce4bcf267d09fed7e3f834ac1ac40966d9d/src/EFCore/Diagnostics/WarningsConfiguration.cs)